File: automaton.py
class Automaton:
    def __init__(self,id,nome,x,y,label,transition,visitado):
        self.id = id
        self.nome = nome
        self.x = x
        self.y = y
        self.label = label
        self.transition = transition
        self.visitado = visitado
        
class AutomatonFactory:
    def create_automaton(self, automaton):
        return Automaton(
            automaton.id,
            automaton.nome,
            automaton.x,
            automaton.y,
            automaton.label,
            automaton.transition,
            automaton.visitado,
        )



def pegaAutomato(automaton):
        return Automaton(
            automaton.id,
            automaton.nome,
            automaton.x,
            automaton.y,
            automaton.label,
            automaton.transition,
            automaton.visitado,
        )

File: projeto-ia/Assets/projeto-ia/Automaton.cs
namespace AutomatonNameSpace
{

    using System.Collections.Generic;
    public class Automaton
    {
        public int id;
        public float x, y, label;
        public string nome;
        public List<int[]> transition;
        public bool visitado;
        public Automaton(int id, string nome, float x, float y, float label)
        {
            this.id = id;
            this.nome = nome;
            this.x = x;
            this.y = y;
            this.label = label;
            transition = new();
            visitado = false;
        }
        public Automaton(Automaton automaton)
        {
            id = automaton.id;
            nome = automaton.nome;
            x = automaton.x;
            y = automaton.y;
            label = automaton.label;
            transition = automaton.transition;
            visitado = automaton.visitado;
        }
    }
}